ksqldb-functional-tests/src/test/resources/query-validation-tests/source-stream.json

```diff
@@ -0,0 +1,48 @@
+{
+  "tests": [
+    {
+      "name": "create source stream",
+      "statements": [
+        "CREATE SOURCE STREAM INPUT (K STRING KEY, text STRING) WITH (kafka_topic='test_topic', value_format='DELIMITED');",
+        "CREATE STREAM OUTPUT AS select * from INPUT;"
+      ],
+      "inputs": [
+        {"topic": "test_topic", "value": "a1"},
+        {"topic": "test_topic", "value": "b1"},
+        {"topic": "test_topic", "value": "B2"}
+      ],
+      "outputs": [
+        {"topic": "OUTPUT", "value": "a1"},
+        {"topic": "OUTPUT", "value": "b1"},
+        {"topic": "OUTPUT", "value": "B2"}
+      ],
+      "post": {
+        "sources": [
+          {
+            "name": "INPUT",
+            "type": "stream",
+            "schema": "K STRING KEY, TEXT STRING",
+            "isSource": true
+          },
+          {
+            "name": "OUTPUT",
+            "type": "stream",
+            "schema": "K STRING KEY, TEXT STRING",
+            "isSource": false
+          }
+        ]
+      }
+    },
+    {
+      "name": "rejects INSERT statements for source streams",
+      "statements": [
+        "CREATE SOURCE STREAM INPUT (K STRING KEY, text STRING) WITH (kafka_topic='test_topic', value_format='DELIMITED');",
+        "INSERT INTO INPUT VALUES ('a', 'b');"
+      ],
+      "expectedException": {
+        "type": "io.confluent.ksql.util.KsqlException",
+        "message": "Cannot insert values into read-only stream: INPUT"
+      }
+    }
+  ]
```

Comment:
Could you also add tests with INSERT INTO SELECT?

Reply:
Added